pkg/domain/filters/containers.go

```diff
@@ -212,6 +212,24 @@ func GenerateContainerFilterFuncs(filter string, filterValues []string, r *libpo
 	case "network":
 		return func(c *libpod.Container) bool {
 			networks, _, err := c.Networks()
+
+			networkMode := c.NetworkMode()
+			// support docker like `--filter network=container:<IDorName>`
+			// check if networkMode is configured as `container:<ctr>`
+			// peform a match against filter `container:<IDorName>`
+			// networks is already going to be empty if `container:<ctr>` is configured as Mode
+			if strings.Contains(networkMode, "container:") {
+				networkModeContainerID := strings.SplitN(networkMode, ":", 2)[1]
+				for _, val := range filterValues {
+					if strings.Contains(val, "container:") {
+						filterNetworkModeIDorName := strings.SplitN(val, ":", 2)[1]
+						filterID, _ := r.LookupContainerID(filterNetworkModeIDorName)
+						if filterID == networkModeContainerID {
+							return true
+						}
+					}
+				}
```

Comment:
You can return false here, if the network mode is `container:` we do not have to check network names.

Reply:
resolved in latest commit.